google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/MessageDispatcher.java

```diff
@@ -372,13 +373,46 @@ void processReceivedMessages(List<ReceivedMessage> messages) {
         // totally expire so that pubsub service sends us the message again.
         continue;
       }
-      outstandingBatch.add(new OutstandingMessage(message, ackHandler));
-      pendingReceipts.add(ackRequestData);
+      if (this.exactlyOnceDeliveryEnabled.get()) {
```

Comment:
Any way to add a unit test for the new behavior?

Reply:
Added